tests/unit/test_config.py

```diff
@@ -89,6 +93,109 @@ def test_can_chain_chalice_stage_values():
     assert not prod.manage_iam_role
 
 
+def test_can_chain_function_values():
+    disk_config = {
+        'lambda_timeout': 10,
+        'stages': {
+            'dev': {
+                'lambda_timeout': 20,
+                'lambda_functions': {
+                    'api_handler': {
+                        'lambda_timeout': 30,
+                    }
+                }
+            }
+        }
+    }
+    c = Config(chalice_stage='dev',
+               config_from_disk=disk_config)
+    assert c.lambda_timeout == 30
+
+
+def test_can_create_scope_obj_with_new_function():
+    disk_config = {
+        'lambda_timeout': 10,
+        'stages': {
+            'dev': {
+                'lambda_timeout': 20,
+                'lambda_functions': {
+                    'api_handler': {
+                        'lambda_timeout': 30,
+                    },
+                    'myauth': {
+                        'lambda_timeout': 40,
+                    }
+                }
+            }
+        }
+    }
+    c = Config(chalice_stage='dev', config_from_disk=disk_config)
+    new_config = c.scope(chalice_stage='dev',
+                         function_name='myauth')
+    assert new_config.lambda_timeout == 40
+
+
+@pytest.mark.parametrize('stage_name,function_name,expected', [
+    ('dev', 'api_handler', 'dev-api-handler'),
+    ('dev', 'myauth', 'dev-myauth'),
+    ('beta', 'api_handler', 'beta-api-handler'),
+    ('beta', 'myauth', 'beta-myauth'),
+    ('prod', 'api_handler', 'prod-stage'),
+    ('prod', 'myauth', 'prod-stage'),
+    ('foostage', 'api_handler', 'global'),
```

Comment:
Makes sense. Yeah I think it would be an optional prompt for a deploying to a new stage that is not the very first `chalice deploy`. I feel that the initial `chalice deploy` should work with minimal amount of prompting. I don't think this should be done in this PR though.